lib/backend/syncersv1/updateprocessors/globalnetworkpolicyprocessor.go

```diff
@@ -43,16 +45,24 @@ func convertGlobalNetworkPolicyV2ToV1Value(val interface{}) (interface{}, error)
 	if !ok {
 		return nil, errors.New("Value is not a valid GlobalNetworkPolicy resource value")
 	}
-	return convertGlobalPolicyV2ToV1Spec(v3res.Spec)
-}
 
-func convertGlobalPolicyV2ToV1Spec(spec apiv3.GlobalNetworkPolicySpec) (*model.Policy, error) {
+	spec := v3res.Spec
+	selector := spec.Selector
+
+	nsSelector := spec.NamespaceSelector
+	if nsSelector != "" {
+		selector = PrefixAndAppendSelector(selector, nsSelector, conversion.NamespaceLabelPrefix)
```

Comment:
Hm, looking at this now, I'm wondering if we need an equivalent for the SA selector as well. I bet that is an existing bug in the rule logic as well.

e.g., `serviceAccountSelector: "all()"` should end up as `has(projectcalico.org/serviceaccount)`.

Reply:
Fixed